File: solution/backtracking/15652/main.cpp
// Authored by : tony9402
// Co-authored by : -
// Link : http://boj.kr/a90bad59d92f4f87bbb056c9e643b330
#include<bits/stdc++.h>

using namespace std;

int choose[10], N, M;

void dfs(int pre, int cnt) {
    if(cnt == M) {
        for(int i=0;i<M;i++) {
            cout << choose[i] << ' ';
        }
        cout << '\n';
        return;
    }

    for(int i=pre;i<=N;i++) {
        choose[cnt] = i;
        dfs(i, cnt + 1);
    }
}

int main(){
    ios::sync_with_stdio(false);
    cin.tie(0);

    cin >> N >> M;
    dfs(1, 0);

    return 0;
}

File: solution/backtracking/15652/main.py
# Authored by : tony9402
# Co-authored by : -
# Link : http://boj.kr/2a9874489cb74d7babd30d70501acff9
import sys

def input():
    return sys.stdin.readline().rstrip()

N, M = map(int, input().split())
choose = [ 0 for _ in range(10) ]

def dfs(idx, cnt):
    global N, M
    if cnt == M:
        for idx in range(cnt):
            print(choose[idx], end=' ')
        print()
        return

    for i in range(idx, N + 1):
        choose[cnt] = i
        dfs(i, cnt + 1)

dfs(1, 0)


